packages/twenty-front/src/pages/settings/integrations/SettingsIntegrationNewDatabaseConnection.tsx

```diff
@@ -15,17 +18,45 @@ import { getSettingsPagePath } from '@/settings/utils/getSettingsPagePath';
 import { AppPath } from '@/types/AppPath';
 import { SettingsPath } from '@/types/SettingsPath';
 import { H2Title } from '@/ui/display/typography/components/H2Title';
+import { useSnackBar } from '@/ui/feedback/snack-bar-manager/hooks/useSnackBar';
 import { SubMenuTopBarContainer } from '@/ui/layout/page/SubMenuTopBarContainer';
 import { Section } from '@/ui/layout/section/components/Section';
 import { Breadcrumb } from '@/ui/navigation/bread-crumb/components/Breadcrumb';
 import { useIsFeatureEnabled } from '@/workspace/hooks/useIsFeatureEnabled';
+import { CreateRemoteServerInput } from '~/generated-metadata/graphql';
 
 const newConnectionSchema = settingsIntegrationPostgreSQLConnectionFormSchema;
 
+const createRemoteServerInputSchema = newConnectionSchema
+  .extend({
+    foreignDataWrapperType: z.string().min(1),
+  })
+  .transform<CreateRemoteServerInput>((values) => ({
+    foreignDataWrapperType: values.foreignDataWrapperType,
+    foreignDataWrapperOptions: {
+      dbname: values.dbname,
+      host: values.host,
+      port: values.port,
+    },
+    userMappingOptions: {
+      password: values.password,
+      user: values.username,
+    },
+  }));
+
 type SettingsIntegrationNewConnectionFormValues = z.infer<
   typeof newConnectionSchema
 >;
 
+const getForeignDataWrapperType = (databaseKey: string) => {
+  switch (databaseKey) {
+    case 'postgresql':
+      return 'postgres_fdw';
+    default:
+      return null;
+  }
+};
+
```

Comment:
This would be useful in `useGetDatabaseConnections.ts` as well I think, maybe we could move it to its own 'utils' file?